firebase-dynamic-links/ktx/src/test/kotlin/com/google/firebase/dynamiclinks/ktx/DynamicLinksTests.kt

```diff
@@ -0,0 +1,98 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.dynamiclinks.ktx
+
+import android.net.Uri
+import com.google.common.truth.Truth.assertThat
+import com.google.firebase.FirebaseApp
+import com.google.firebase.FirebaseOptions
+import com.google.firebase.dynamiclinks.DynamicLink
+import com.google.firebase.ktx.Firebase
+import com.google.firebase.dynamiclinks.FirebaseDynamicLinks
+import com.google.firebase.ktx.app
+import com.google.firebase.ktx.initialize
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.RuntimeEnvironment
+
+const val APP_ID = "APP_ID"
+const val API_KEY = "API_KEY"
+
+const val EXISTING_APP = "existing"
+
+abstract class BaseTestCase {
+    @Before
+    fun setUp() {
+        Firebase.initialize(
+                RuntimeEnvironment.application,
+                FirebaseOptions.Builder()
+                        .setApplicationId(APP_ID)
+                        .setApiKey(API_KEY)
+                        .setProjectId("123")
+                        .build()
+        )
+
+        Firebase.initialize(
+                RuntimeEnvironment.application,
+                FirebaseOptions.Builder()
+                        .setApplicationId(APP_ID)
+                        .setApiKey(API_KEY)
+                        .setProjectId("123")
+                        .build(),
+                EXISTING_APP
+        )
+    }
+
+    @After
+    fun cleanUp() {
+        FirebaseApp.clearInstancesForTest()
+    }
+}
+
+@RunWith(RobolectricTestRunner::class)
+class DynamicLinksTests : BaseTestCase() {
+
+    @Test
+    fun `Firebase#dynamicLinks should delegate to FirebaseDynamicLinks#getInstance()`() {
+        assertThat(Firebase.dynamicLinks).isSameInstanceAs(FirebaseDynamicLinks.getInstance())
+    }
+
+    @Test
+    fun `Firebase#dynamicLinks should delegate to FirebaseDynamicLinks#getInstance(FirebaseApp)`() {
+        val app = Firebase.app(EXISTING_APP)
+        assertThat(Firebase.dynamicLinks(app))
+                .isSameInstanceAs(FirebaseDynamicLinks.getInstance(app))
+    }
+
+    @Test
+    fun `androidParameters type-safe builder extension works`() {
+        val fallbackUri = Uri.parse("https://example.com")
+
+        val parametersKTX = androidParameters {
+            setMinimumVersion(16)
+            setFallbackUrl(fallbackUri)
+        }
+
+        val parameters = DynamicLink.AndroidParameters.Builder()
+                .setMinimumVersion(16)
+                .setFallbackUrl(Uri.parse("https://example.com"))
+                .build()
+
+        assert(parameters.equals(parametersKTX))
```

Comment:
Note to reviewer: Not sure if this is the best way to test the builder extensions. My idea was to add an equals() method to DynamicLink.AndroidParameters and pass it to assert(). Let me know if we should use a different approach here.

Reply:
I'd really like not to change the current behavior of fdl, i.e. changing equals may in theory break existing code out there.

I have 2 suggestions, none of which are ideal as they rely on implementation details of the class, but should get the job done:
* Move the test into the `com.google.firebase.dynamiclinks` package this will give you access to `DynamicLink.AndroidParameters#parameters`(package private), you can make sure that the ktx builder has correctly populated the bundle
* Have a test that creates a DynamicLink with androidParameters and parses `getUri()` to make sure android params are correctly set

Reply:
I like the latter suggestion better. I'll give it a try after modifying the builder functions as suggested in the comments below.